backend/fms_core/viewsets/_utils.py

```diff
@@ -29,6 +29,13 @@ def _prefix_keys(prefix: str, d: Dict[str, Any]) -> Dict[str, Any]:
 def _list_keys(d: Dict[str, Any]) -> Dict[str, Any]:
     return [k  for k, v in d.items()]
 
+def fix_count_derived_samples(queryset: QuerySet) -> QuerySet:
+    # https://stackoverflow.com/a/69031027
+    return queryset.annotate(count_derived_samples=Subquery(
+        DerivedBySample.objects
+            .filter(sample=OuterRef("pk"))
+            .annotate(my_count=Func(F('id'), function='Count'))
```

Comment:
why not just Count("id") ? i think it actually does the same as what you are doing.